scripts/concatenate_files.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+import re
+import yaml
+
+docs_root = "./docs"
+
+
+def concat(of, level, doc_path):
+    full_path = f"{docs_root}/{doc_path}.md"
+    if not os.path.exists(full_path):
+        full_path = f"{docs_root}/{doc_path}/index.md"
+
+    with open(full_path, "r") as doc_file:
+        doc_text = doc_file.read()
+        # parse YAML header
+        yaml_match = re.match("^---$((.|\n)*?)^---$", doc_text, flags=re.MULTILINE).groups(1)
+        if yaml_match:
+            header = yaml.safe_load(yaml_match[0])
+            of.write(f"""{"#"*level} {header["title"]}""")
+            # strip the YAML header
+            doc_text = re.sub("^---$((.|\n)*)?^---$", "", doc_text, flags=re.MULTILINE)
+        # use relative path for images
+        doc_text = doc_text.replace("](/images", "](images")
+        of.write(doc_text)
+
+
+f = open("./_data/menu_docs_dev.json", "r")
+
+of = open("concat.md", "w")
+
+data = json.load(f)
+
+documentation_main_level_pages = [x for x in data["docsmenu"] if x["page"] == "Documentation"][0]["mainfolderitems"]
+
+for main_level_page in documentation_main_level_pages:
+    main_title = main_level_page["page"]
+
+    main_url = main_level_page.get("url")
+    main_slug = main_level_page.get("slug")
+
+    if main_url:
+        print(f"- {main_url}")
+        concat(of, 1, f"{main_url}")
+
+    if not main_slug:
+        continue
+
+    print(f"- {main_slug}")
+    for subfolder_page in main_level_page["subfolderitems"]:
+        subfolder_url = subfolder_page.get("url")
+        if subfolder_url:
+            print(f"  - {main_slug}/{subfolder_url}")
+            concat(of, 2, f"{main_slug}/{subfolder_url}")
+
+        subfolder_slug = subfolder_page.get("slug")
+        if not subfolder_slug:
+            continue
+
+        print(f"  - {main_slug}/{subfolder_slug}")
+        for subsubfolder_page in subfolder_page["subsubfolderitems"]:
+            subsubfolder_url = subsubfolder_page.get("url")
+
+            print(f"    - {main_slug}/{subfolder_slug}/{subsubfolder_url}")
+            concat(of, 3, f"{main_slug}/{subfolder_slug}/{subsubfolder_url}")
```

Comment:
Maybe just make this into a recursive function instead of doing subsub....?

Reply:
I found that the three-levels are sufficiently different so that a recursive function would be quite complicated.